lib/wcc/auth/devise.rb

```diff
@@ -1,22 +1,37 @@
-require "wcc/auth"
-require "devise"
+# frozen_string_literal: true
+
+require 'wcc/auth'
+require 'devise'
 
 module WCC
   module Auth
     module Devise
-      autoload :WatermarkCallbacksController, 'wcc/auth/devise/watermark_callbacks_controller'
+      autoload :WatermarkCallbacksController,
+               'wcc/auth/devise/watermark_callbacks_controller'
     end
+
+    SetupOAuth = lambda { |env|
+      request = Rack::Request.new(env)
+
+      env['omniauth.strategy'].options[:client_id] = config.app_id
+      env['omniauth.strategy'].options[:client_secret] = config.app_secret
+      auth_params = config.authorize_params.merge(
+        request.params['authorize_params'] || {}
```

Comment:
Is this not a security vulnerability?  What all might I be able to configure here just by passing my own `?authorize_params=`?